Packages/com.unity.inputsystem/InputSystem/Controls/InputControl.cs

```diff
@@ -952,8 +952,9 @@ internal void SetOptimizedControlDataTypeRecursively()
         // This is mainly to AxisControl fields being public and capable of changing at any time even if we were not anticipated such a usage pattern.
         // Also it's not clear if InputControl.stateBlock.format can potentially change at any time, likely not.
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        // Only do this check in development builds and editor in hope that it will be sufficient to catch any misuse during development.
-        [Conditional("DEVELOPMENT_BUILD"), Conditional("UNITY_EDITOR")]
+        // Only do this check in and editor in hope that it will be sufficient to catch any misuse during development.
+        // It is not done in debug builds because it has a performance cost and it will show up when profiled.
```

Comment:
I am slightly confused here, why would it be bad to have this in a debug build? Debug mode code shouldn't be profiled unless some kind of rough ballpark number is of interest? I think checking in editor-only might be fine, but I am confused about caring about debug build and profiling numbers. I mean many non-UNity tools shows you many red warnings when trying to profile or benchmark something in a debug build.

Reply:
You can only use the Unity profiler with development builds. See https://docs.unity3d.com/Manual/profiler-profiling-applications.html

Reply:
Seems the only way to do release builds profiling on Unity is by using native profilers. I didn't not look into those and the reported issue uses the Unity profiler as well.

Reply:
OK, I would recommend taking a look with e.g. Superluminal to make sure we understand behavior in release build.